### ISSUE TITLE
set EPOLLONESHOT when inserts a file descriptor in epoll ?

### ISSUE THREAD

**Issue by cssivision** [stjepang/polling#11]
epoll use level-triggered interface as default,  epoll_wait will always report for EPOLLHUP | EPOLLERR events.

current implementation, insert function register file description with events equal to 0.
```rust
pub fn insert(&self, fd: RawFd) -> io::Result<()> {
        log::debug!("insert: epoll_fd={}, fd={}", self.epoll_fd, fd);

        // Put the file descriptor in non-blocking mode.
        let flags = syscall!(fcntl(fd, libc::F_GETFL))?;
        syscall!(fcntl(fd, libc::F_SETFL, flags | libc::O_NONBLOCK))?;

        // Register the file descriptor in epoll.
        let mut ev = libc::epoll_event {
            events: 0,
            u64: 0u64,
        };
        syscall!(epoll_ctl(self.epoll_fd, libc::EPOLL_CTL_ADD, fd, &mut ev))?;

        Ok(())
    }

```

after inserting in epoll, EPOLLHUP | EPOLLERR triggered,  and no read/write call on this file description. the main_loop of Reactor will run forever.

**Comment by stjepang** [comment]
That's a good point, thanks for reporting the issue! Submitted a fix here: https://github.com/stjepang/polling/pull/12

**Comment by stjepang** [comment]
Published v0.1.7
